keyvault/client/src/main/java/com/azure/keyvault/SecretClient.java

```diff
@@ -0,0 +1,426 @@
+    }
+
+    /**
+     * The get deleted secret operation returns the secrets that have been deleted for a vault enabled for soft-delete.
+     * This operation requires the {@code secrets/list} permission.
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <pre>
+     * //Assuming secret is deleted on a soft-delete enabled key vault.
+     * DeletedSecret deletedSecret = secretClient.getDeletedSecret("secretName").value();
+     * System.out.println(String.format("Deleted Secret with recovery Id %s", deletedSecret.recoveryId()));
+     * </pre>
+     *
+     * @param name The name of the deleted secret.
+     * @return A {@link Response} whose {@link Response#value()} contains the deleted {@link DeletedSecret}.
+     * @throws com.azure.common.exception.ServiceRequestException when a deleted secret with {@code name} doesn't exist in the key vault.
+     */
+    public Response<DeletedSecret> getDeletedSecret(String name) {
+        return service.getDeletedSecret(vaultEndpoint, name, API_VERSION, ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE).block();
+    }
+
+    /**
+     * The purge deleted secret operation removes the secret permanently, without the possibility of recovery.
+     * This operation can only be enabled on a soft-delete enabled vault. This operation requires the {@code secrets/purge} permission.
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <pre>
+     * //Assuming secret is deleted on a soft-delete enabled key vault.
+     * secretClient.purgeDeletedSecret("deletedSecretName");
+     * </pre>
+     *
+     * @param name The name of the secret.
+     * @return A {@link VoidResponse}.
+     * @throws com.azure.common.exception.ServiceRequestException when a deleted secret with {@code name} doesn't exist in the key vault.
+     */
+    public VoidResponse purgeDeletedSecret(String name) {
+        return service.purgeDeletedSecret(vaultEndpoint, name, API_VERSION, ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE).block();
+    }
+
+    /**
+     * Recovers the deleted secret in the key vault to its latest version and can only be performed on a soft-delete enabled vault.
+     * This operation requires the {@code secrets/recover} permission.
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <pre>
+     * //Assuming secret is deleted on a soft-delete enabled key vault.
+     * Secret recoveredSecret =  secretClient.recoverDeletedSecret("deletedSecretName").value();
+     * System.out.println(String.format("Recovered Secret with name %s", recoveredSecret.name()));
+     * </pre>
+     *
+     * @param name The name of the deleted secret to be recovered.
+     * @return A {@link Response} whose {@link Response#value()} contains the recovered {@link Secret}.
+     * @throws com.azure.common.exception.ServiceRequestException when a deleted secret with {@code name} doesn't exist in the key vault.
```

Comment:
Be sure to circle-back with improved exceptions

Reply:
added improved exceptions